velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp

```diff
@@ -1243,6 +1282,42 @@ TEST_F(MinMaxByNTest, global) {
       }),
   });
 
+  testAggregations(
+      {data}, {}, {"min_by(c0, c1, 3)", "max_by(c0, c1, 4)"}, {expected});
+
+  // bool type of comparison
+  data = makeRowVector({
+      makeFlatVector<int32_t>({1, 2}),
+      makeFlatVector<bool>({true, false}),
+  });
+
+  expected = makeRowVector({
+      makeArrayVector<int32_t>({
+          {2},
+      }),
+      makeArrayVector<int32_t>({
+          {1},
+      }),
+  });
+
+  testAggregations(
```

Comment:
Can you do a bool type of comparison test with array length more than 1 too for completion sake ?

Reply:
I have tried, but we cannot do results assertion easily cause results are non-deterministic when comparison type is bool, and MinMaxN use priority_queue as heap to store valid values actually.